deploy/test/config/k8s/secrets-access-role.yaml

```diff
@@ -0,0 +1,28 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ ID }}secrets-provider-service-account
+  namespace: {{ APP_NAMESPACE_NAME }}
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: {{ ID }}secrets-provider-role
+  namespace: {{ APP_NAMESPACE_NAME }}
+rules:
+  - apiGroups: [""]
+    resources: ["secrets"]
+    verbs: [ "get", "update" ]
```

Comment:
Is that a duplication of the right necessary for secrets provider?
Can we reuse one of the existing manifests?

Reply:
no for the same reason explained before. The process of generating the files are different between init tests and helm tests. Didn't want to take on the refactor in this PR

Reply:
where is the difference between init container and pod of secret provider seen here ?